index.js

```diff
@@ -13,47 +14,123 @@ class Executor {
     }
 
     /**
-     * Start the executor
+     * Validate the config for _start method
      * @method start
      * @param {Object} config               Configuration
      * @param {String} config.buildId       Build id
      * @param {String} config.jobId         Job id
      * @param {String} config.pipelineId    Pipeline id
      * @param {String} config.container     Container
      * @param {String} config.scmUrl        Scm url
+     * @param {Function} callback           Function to call when done
+     */
+    start(config, callback) {
+        const schema = Joi.object().keys({
+            buildId: Joi.string().required(),
+            jobId: Joi.string().required(),
+            pipelineId: Joi.string().required(),
+            container: Joi.string().required(),
+            scmUrl: Joi.string().required()
+        });
+
+        const result = Joi.validate(config, schema);
+
+        if (result.error) {
+            return callback(result.error);
+        }
+
+        return this._start(config, callback);
+    }
+
+    /**
+     * Start the executor
+     * @method _start
+     * @param {Object} config               Configuration
+     * @param {String} config.buildId       Build id
+     * @param {String} config.jobId         Job id
+     * @param {String} config.pipelineId    Pipeline id
+     * @param {String} config.container     Container
+     * @param {String} config.scmUrl        Scm url
      * @param {Function} callback
      */
-    start() {
-        console.error('start is not implemented');
-        throw new Error('not implemented');
+    _start(config, callback) {
+        callback(new Error('not implemented'));
     }
 
     /**
-     * Stop the executor
+     * Validate the config for _stop method
      * @method stop
      * @param {Object} config               Configuration
      * @param {String} config.buildId       Build id
```

Comment:
Same for `start`, we only need buildId and maybe container. k8s takes all the extra fields anyway (for labels) but it's not necessary.